dopamine/discrete_domains/atari_lib.py

```diff
@@ -74,7 +80,7 @@ def create_atari_environment(game_name=None, sticky_actions=True):
   assert game_name is not None
   game_version = 'v0' if sticky_actions else 'v4'
   full_game_name = '{}NoFrameskip-{}'.format(game_name, game_version)
-  env = gym.make(full_game_name)
+  env = gym.make(full_game_name, mode=game_mode, difficulty=game_difficulty)
```

Comment:
Ddoes gym.make() work with mode and difficulty, irrespective of the domain? Or should we add an assert that game_name needs to be a valid Atari 2600 environment?

Reply:
It only works with the Atari 2600 environment. Isn't this what `create_atari_environment` is for?

Reply:
Correct. But if someone passes in a game_name that does not correspond to a valid Atari 2600 game, what will the error message be?

Reply:
Ahh, I see, so Gym supports arbitrary kwargs for `gym.make` (see https://github.com/openai/gym/pull/1301). If the environment isn't found it will just throw an error saying the environment isn't registered regardless of the kwargs passed.

It doesn't fail differently because we providing `mode` and `difficulty` to `gym.make`.

Reply:
I thought it would be easier to check if the environment string was registered as an Atari environment. It turns out it wasn't as easy as I thought. I ended up submitting a PR to gym to make this easier https://github.com/openai/gym/pull/1597.

Hopefully, your concern is now addressed in this latest commit. It's also worth noting we could now deal with frame skipping through kwargs instead of appending FrameSkip onto the environment ID. This might be a nicer way to deal with it. I can make this change if you'd like.